src/lib/Util.hs

```diff
@@ -174,13 +175,20 @@ findReplace old new s@(x:xs) =
 traverseFun :: Traversable t => (a -> s -> (b, s)) -> t a -> s -> (t b, s)
 traverseFun f xs s = runState (traverse (asState . f) xs) s
 
-asState :: (s -> (a, s)) -> State s a
-asState f = do
+traverseFunM :: (Monad m, Traversable t)
+  => (a -> s -> m (b, s)) -> t a -> s -> m (t b, s)
+traverseFunM f xs s = runStateT (traverse (asStateT . f) xs) s
```

Comment:
Wouldn't `scanM` be a better name?

Reply:
Sounds reasonable to me. Should I change `traverseFun` to `scan` too?